component/azstorage/azstorage.go

```diff
@@ -207,7 +207,7 @@ func (az *AzStorage) ListContainers() ([]string, error) {
 func (az *AzStorage) CreateDir(options internal.CreateDirOptions) error {
 	log.Trace("AzStorage::CreateDir : %s", options.Name)
 
-	err := az.storage.CreateDirectory(internal.TruncateDirName(options.Name))
+	err := az.storage.CreateDirectory(internal.TruncateDirName(options.Name), options.IsNoneMatchEtagEnabled)
```

Comment:
If this flag is enabled, we always hardcode the filter to "*" which means in create directory we are only asking to check if the directory exists or not and the flag controls whether to create directory forcefully or fail if it exists then name of the flag is not depicting that correctly.